packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -286,6 +286,69 @@ export class ProjectService extends BaseService {
         this.groupsModel = groupsModel;
     }
 
+    static validateProjectCreationPermissions(
+        user: SessionUser,
+        data: CreateProject,
+    ) {
+        if (!data.type) {
+            throw new ParameterError('Project type must be provided');
+        }
+
+        if (data.type === ProjectType.DEFAULT && data.upstreamProjectUuid) {
+            throw new ParameterError(
+                'upstreamProjectUuid must not be provided for default projects',
+            );
+        }
+
+        switch (data.type) {
+            case ProjectType.DEFAULT:
+                // checks if user has permission to create project on an organization level
+                if (
+                    user.ability.can(
+                        'create',
+                        subject('Project', {
+                            organizationUuid: user.organizationUuid,
+                            type: ProjectType.DEFAULT,
+                        }),
+                    )
+                ) {
+                    return true;
+                }
+
+                throw new ForbiddenError();
+
+            case ProjectType.PREVIEW:
+                // if the upstream project is not provided, user must have access to create project ONLY on an organization level
+                // if the upstream project is provided, user must have access to the upstream project level OR on an organization level
+                if (
+                    user.ability.can(
+                        'create',
+                        subject('Project', {
+                            upstreamProjectUuid: data.upstreamProjectUuid,
+                            type: ProjectType.PREVIEW,
+                        }),
+                    ) ||
+                    user.ability.can(
+                        'create',
+                        subject('Project', {
+                            organizationUuid: user.organizationUuid,
```

Comment:
I'm not 100% sure about this, but I think this condition is kind of wrong. 

A user who doesn't have access  to the project, can still create a preview with an upstreamProjectUuid (aka copying content) if you can create a project at the org level. 

This might still work because give our project role inheritance and our ability config , if you are a developer on the org, you are a dev on all the projects. 
But This is wrong from the permission point of view. Does that make sense ? 

An easy fix for this would be to simply check the ability inside an if like 

```
if (data.upstreamProjectUuid) {
  // check project ability
} else {
// check organization ability
}
```

Reply:
I had it set up as you showed in your code example, but I had to change it because of exactly this:

> This might still work because give our project role inheritance and our ability config , if you are a developer on the org, you are a dev on all the projects.

There are developers who inherit permissions from the organization level. If you are a developer at the organization level and a viewer at the project level, you are still considered a developer on the project because the highest role takes precedence.